### ISSUE TITLE
Decide what media URL's should look like

### ISSUE THREAD

**Issue by karindalziel** [CDRH/data#113]
right now we have a media_base URL, which typically then built a url like `cdrhmedia.unl.edu/projectname/image/size/123.jpg`

With the image server, we'll have URL's more like:

https://cdrhmedia.unl.edu/iiif/2/cather%2Fcat.let0019.001.jpg/full/!1000,1000/0/default.jpg

So we need to pass in the project name as well. 

It could be that we just use the shortname to build that part of the URL, but I wanted to talk it through first. (Right now, the cather shortname is `cather-complete-letters` but I have names the IIIF folder "cather" - we'll need to sandardize)
We could also have a different variable for "iiif folder name" or something. 

Here's an example of where I build the URL: https://github.com/Willa-Cather-Archive/complete-letters/blob/iiif/scripts/tei.p5.xsl#L849

**Comment by jduss4** [comment]
Right now we already have to pass in the project name for most of the $media_base stuff, as far as I am aware.
https://github.com/CDRH/data/blob/master/scripts/xslt/cdrh_to_html/lib/cdrh.xsl#L144

But as I have mentioned to Karin, I have been regarding the $media_base as the absolute most basic unit of URL.  Like, `cdrhmedia.unl.edu` or `cdrhdev1.unl.edu/media` so then you have to build the rest of the URL in a template / method / etc.

**Comment by techgique** [comment]
I think Jess's approach to URLs fits with what I would expect from variables named "base", just when thinking about names used by other web software. So I'm not sure if y'all are thinking about using a different URL with more of the path for the project, but I'd suggest making a new variable with a different name, otherwise it wouldn't match programmers' reasonable expectations for a "base" URL. Maybe something like `project_iiif`? I thought `iiif_base`, but I would expect a variable named that not to have the project-specific parts too. Don't know if any of that helps :man_shrugging:

**Comment by jduss4** [comment]
`$iiif_path`?

**Comment by jduss4** [comment]
I think if things are in subdirectories then we're going to have to put that in an override rather than a config variable, not only for flexibility of altering it for various aspects of a document (example: "type 'text' == transmissisippi%2Fscrapbooks' and everything else == 'transmississippi%2Fmemorabilia' ") but because of needing to encode directory navigation with the IIIF %2F thing.

**Comment by karindalziel** [comment]
I was thinking of a situation like this:

- whitman_scribal has images in whitman/scribal
- whitman_marginalia has images in whitman/marginalia

Rather than having to override both templates, they could use the same one if the the `iif_segment` was, for instance, `whitman%2Fscribal`

**Comment by jduss4** [comment]
yeah, technically we could probably just put whitman logic in the whitman-scripts xslt, since it's common to all of them.  but I take your point!
